source/upgrading-django.rst

```diff
@@ -80,3 +81,88 @@ of the package I ended up with, and update the requirements file
+   If there is anything from the notes that clearly needs to be
+   updated (a certain function is deprecated, etc.), make those
+   updates to the code. Don't worry about catching everything
+   (sometimes there's not anything that sticks out from the notes).
+
+2. Determine if our code generates any deprecation warnings
+   by running the test suite. Use the ``-W`` command line argument
+   to show the deprecation warnings. Adding ``once`` after ``-W``
+   will only show the first occurrence of each warning in order
+   to make seeing the warnings more manageable::
+
+       python -Wonce manage.py test
+
+   A deprecation warning will look like::
+
+       /Users/username/projects/coolproject/coolapp/models.py:65: RemovedInDjango20Warning: on_delete will be a required arg for ForeignKey in Django 2.0. Set it to models.CASCADE on models and in existing migrations if you want to maintain the current default behavior. See https://docs.djangoproject.com/en/1.11/ref/models/fields/#django.db.models.ForeignKey.on_delete
+       category = models.ForeignKey(Category)
+
+3. If our code has deprecation warnings, then make the changes
+   there. If one of our dependencies has deprecation warnings,
+   then look into whether a newer version of the dependency is
+   available, and try to upgrade that first. It may be helpful
+   to make a few or just 1 change at a time, rather than trying
+   to do everything at once. Assuming that thorough tests have
+   been written for our application, feel free to run the test
+   suite after each set of changes to make sure that everything
+   still runs. Sometimes things will break unexpectedly and you
+   will need to determine why. Googling the error messages is
+   oftentimes helpful in such a situation.
+
+   Note: if you upgrade a dependency, make sure you update
+   ``requirements.txt`` with the newer version of the dependency,
+   as well as newer versions of sub-dependencies. For example,
+   if you update the ``celery`` dependency, and this upgrade
+   also upgrades the ``pytz`` sub-dependency, then pin down both
+   the ``celery`` version and the ``pytz`` version in the
+   appropriate requirements file.
+
+   It may also be a good idea to review our dependencies even
+   if they don't generate deprecation warnings. Feel free to
+   upgrade dependencies to newer versions and to run the test
+   suite to make sure the upgrade didn't break anything, and make
+   sure to update the version number in appropriate requirements
+   file. You can see which dependencies have newer releases
+   by running::
+
+       pip list --outdated
+
+   and see the same list in a column format with::
+
+       pip list --outdated --format=columns
+
+4. Once all of the deprecation warnings have been resolved,
+   update the requirements file with the new version of Django
+   and run ``pip install -r requirements.txt``.
+
+5. Try to run the server locally with ``python manage.py runserver``,
```

Comment:
Most of our sites need to run `npm somethin somethin` to work, but I think the dev should understand that. (or you could remove the `manage.py` command and leave it as an exercise for the reader to figure out how to run the server locally.)

Reply:
Heh, I remember the '... will be left as an exercise for the reader' text from textbooks always being unsatisfactory.